feat: implement min, max and i18n properties

Haprog opened the pull request:
Closes #4.

NOTE: This should be rebased, not squashed. Might be easier to review per commit.

tomivirkki commented on demo/date-time-picker-localization-demos.html:
```diff
@@ -0,0 +1,112 @@
+    <h3>Localizing</h3>
+    <vaadin-demo-snippet id="date-time-picker-localizing" when-defined="vaadin-date-time-picker">
+      <template preserve-content>
+        <vaadin-date-time-picker label="Syntymäpäivä"></vaadin-date-time-picker>
```
nit: birthday is a bit odd label for this when you're also picking the time

Haprog replied:
I couldn't think of a better use case. I was thinking "birth date and time" to specify exact time of birth (like in the "Initial position of date picker" demo).

Would it be ok if I just update the label to "Syntymäpäivä ja -aika"?

tomivirkki replied:
That would be ok 👍

Haprog replied:
Updated.

Haprog commented:
@tomivirkki 
> The DTP's `value` in "Time includes milliseconds" example for some reason no longer has millisecond precision after this PR

Thanks. I'll fix that.

Haprog commented:
@tomivirkki The above bug about millisecond precision has been fixed and a test added for it.

Haprog commented:
> It's quite a large PR so it's hard to notice if there's something not working. But there seems to be some untested functionality. All the tests pass with the following lines commented out: [b6ee4bf](https://github.com/vaadin/vaadin-date-time-picker/commit/b6ee4bfae033592d8feba1043371bc26dd9ddd79)

Thanks. I checked that about half of those lines are not modified in this PR. I'll check at least those lines related to this PR (and maybe open a new issue about the rest).

Haprog commented:
@tomivirkki I checked all the parts you mentioned (commented out) in https://github.com/vaadin/vaadin-date-time-picker/commit/b6ee4bfae033592d8feba1043371bc26dd9ddd79 and pushed a bunch of new commits addressing all of these (even if some of them were not related directly to this PR) imo.

There were few cases of unnecessary code that I just removed. In many cases I added some new test cases to cover the related code so that some tests will fail when commenting those lines out.

Few minor bugs were also found and fixed while doing this.

Few of the code chunks did not need any action imo.

Here's some additional comments on specific lines that were not changed or not fully covered by tests:
- "early return": Lines 364-366, 397-399, 568-570 are just early return to make the code nicer looking (imo) by avoiding unnecessary indentation/nesting. These conditions could all be inverted and then move all the rest of the code in the method into the block to get rid of the extra `return;` but I think this way is nicer. It makes no sense to test the empty return statement in these cases imo.
- Lines 639,645,648 in `__stepSegment`: I improved the test coverage of `__stepSegment` slightly but there are still few lines not covered there. The method could be refactored and shortened so that it only has the cases that are really needed in the date time picker now, but I think it's better to keep it as it is as now it is more clear what it does and it's a bit more general purpose. The other branches could be useful in the future. Imo this is good enough now even though not all the return values are covered by tests since those aren't all used atm. and we don't have unit tests. Could add unit tests for this method if you think it makes sense?
- Line 765: the condition around this could be removed so it would always call `__updateTimePickerMinMax()` and all would still work, but this condition is just a minor performance improvement so that the method `__updateTimePickerMinMax` is not called unnecessarily when not needed. I don't think we need to test this, but I'd still keep it as it is now. We could maybe add a test to see that in some case this private method should not be called but I'm not sure if it would make sense to test.

Haprog commented:
I will check the CI errors tomorrow.

Haprog commented:
@tomivirkki Thanks for noticing that bug. I added a test and fix for it.

tomivirkki commented on src/vaadin-date-time-picker.html:
```diff
@@ -292,18 +411,62 @@
+            // propagate the updated value change to custom-field (which doesn't
+            // automatically process programmatically triggered value updates)
+            if (this.__timePicker.value !== oldTimeValue) {
+              this.__triggerCustomFieldValueUpdate();
```
Now in such a case as described in https://github.com/vaadin/vaadin-date-time-picker/pull/19#pullrequestreview-342440923 you'll still get a `value-changed` event with the invalid value immediately followed by one with the corrected one. It should ideally only emit one `value-changed` event with the right value.

Haprog replied:
This has been fixed and the behaviour change a bit (as agreed in Slack).
